sonarcloud.sh

```diff
@@ -0,0 +1,98 @@
+#! /bin/bash
+
+##
+# Copyright Ladislas de Toldi 2018
+# Copyright IBM Corporation 2016,2017,2018
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##
+
+if [[ $TRAVIS_BRANCH != "master" && $TRAVIS_BRANCH != "develop" && $TRAVIS_EVENT_TYPE != "cron" ]]; then
+  echo "Not master, develop or cron build. Skipping code coverage generation."
+  exit 0
+fi
+
+echo ">> Starting Sonar Cloud code coverage analysis..."
+uname -a
+
+SDK=macosx
+DERIVED_DATA=.build/scanner
+xcodebuild -version
+xcodebuild -version -sdk $SDK
+if [[ $? != 0 ]]; then
+  exit 1
+fi
+
+# Determine if there is a custom command for generating xcode project
+source ./Package-Builder/generate-xcodeproj.sh
+
+# Determine if there is a custom command for xcode build (code coverage tests)
+if [ -e ${projectFolder}/.swift-codecov ]; then
+  XCODE_BUILD_CMD=$(cat ${projectFolder}/.swift-codecov)
+else
+  PROJECT="${PROJ_OUTPUT##*/}"
+  SCHEME=$(xcodebuild -list -project $PROJECT | grep --after-context=1 '^\s*Schemes:' | tail -n 1 | xargs)
+  XCODE_BUILD_CMD="xcodebuild -project $PROJECT -scheme $SCHEME -derivedDataPath $DERIVED_DATA -enableCodeCoverage YES clean build test CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO"
+fi
+
+echo ">> Running: $XCODE_BUILD_CMD"
+eval "$XCODE_BUILD_CMD"
+if [[ $? != 0 ]]; then
+  echo ">> Error running: $XCODE_BUILD_CMD"
+  exit 1
+fi
+
+BASH_URL="https://raw.githubusercontent.com/SonarSource/sonar-scanning-examples/master/swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh"
+BASH_BASE="bash <(curl -s $BASH_URL)"
+XCCOV_ARCHIVE_PATH="$DERIVED_DATA/Logs/Test/*.xcresult/*_Test/action.xccovarchive/"
+if [ -z ${SONAR_COVERAGE_REPORT_PATH+x} ]; then
+SONAR_COVERAGE_REPORT_PATH="sonarscanner-coverage-report.xml"
+fi
+
+BASH_CMD="$BASH_BASE $XCCOV_ARCHIVE_PATH > $SONAR_COVERAGE_REPORT_PATH"
+
+echo ">> Running: $BASH_CMD"
+eval "$BASH_CMD"
+if [[ $? != 0 ]]; then
+  echo ">> Error running: $BASH_CMD"
+  exit 1
+fi
+
+if [ -e ${projectFolder}/.swift-sonarcloud ]; then
+  echo ".swift-sonarcloud found, running sonar-scanner with .swift-sonarcloud."
+  SONAR_UPLOAD_CMD=$(cat ${projectFolder}/.swift-sonarcloud)
```

Comment:
Wouldn't this mean that `.swift-sonarcloud` has to contain your SonarCloud login token? Isn't that a bad idea?

Reply:
No because they can still call `sonar-scanner -Dsonar.login=$SONAR_LOGIN_TOKEN` from their file as long as they define in the variable on Travis-CI website.

In addition I would say that this is and "expert" option for people dealing with private repositories or keeping their credentials in private repositories.